ethcore/src/state/account.rs

```diff
@@ -263,6 +281,10 @@ impl Account {
 		if let Some(value) = self.storage_changes.get(key) {
 			return Some(value.clone())
 		}
+		//Use only in memory state for the patched account
+		if self.patched_account {
```

Comment:
That is probably true for others function ('storage_at' will query the triedb for instance), do we need this kind of guard for them too?

I wonder if it wouldn't be ok to use the storage cache and original cache value (init them in 'reset_code_and_storage') and just prevent querying the triedb at 'get_and_cache_storage' ?

Reply:
Actually it is the guard for storage_at, because it directly calls cached_storage_at:

pub fn storage_at(&self, db: &HashDB<KeccakHasher, DBValue>, key: &H256) -> TrieResult<H256> {
		if let Some(value) = self.cached_storage_at(key) {
			return Ok(value);
		}
		Self::get_and_cache_storage(
			&self.storage_root,
			&mut self.storage_cache.borrow_mut(),
			db,
			key)
	}

Reply:
I think 'Self::get_and_cache_storage' will be call due to this function returning 'None' value on missing key, then it will query the triedb (from the 'storage_at' code).